src/wallet/rpcdump.cpp

```diff
@@ -114,6 +112,9 @@ UniValue importprivkey(const UniValue& params, bool fHelp)
     if (params.size() > 2)
         fRescan = params[2].get_bool();
 
+    if (fRescan && fPruneMode)
+        throw JSONRPCError(RPC_WALLET_ERROR, "Rescan is disabled in pruned mode");
```

Comment:
The only improvement I can think of would be to add a comment to the error message that it is possible to pass 'false' as the rescan parameter.

You don't have to worry about translators, RPC error messages aren't translated.